src/spotify_to_tidal/sync.py

```diff
@@ -172,133 +172,159 @@ def get_tracks_from_spotify_playlist(spotify_session: spotipy.Spotify, spotify_p
-  tidal_playlists = get_tidal_playlists_dict(tidal_session)
-  for spotify_playlist in spotify_playlists:
-      results.append( pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists) )
-  return results
-
-def get_playlists_from_spotify(spotify_session: spotipy.Spotify, config):
+    results = []
+    spotify_playlists = asyncio.run(get_playlists_from_spotify(spotify_session, config))
+    tidal_playlists = get_tidal_playlists_dict(tidal_session)
+    for spotify_playlist in spotify_playlists:
+        results.append( pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists) )
+    return results
+
+async def get_playlists_from_spotify(spotify_session: spotipy.Spotify, config):
     # get all the user playlists from the Spotify account
     playlists = []
-    with tqdm(total=1.0) as pbar:
-      pbar.set_description("Loading Spotify playlists")
-      spotify_results = spotify_session.user_playlists(config['spotify']['username'])
-      total = spotify_results['total']
-      exclude_list = set([x.split(':')[-1] for x in config.get('excluded_playlists', [])])
-      while True:
-          pbar.update(len(spotify_results['items'])/total)
-          for spotify_playlist in spotify_results['items']:
-              if spotify_playlist['owner']['id'] == config['spotify']['username'] and not spotify_playlist['id'] in exclude_list:
-                  playlists.append(spotify_playlist)
-          # move to the next page of results if there are still playlists remaining
-          if spotify_results['next']:
-              spotify_results = spotify_session.next(spotify_results)
-          else:
-              break
+    print("Loading Spotify playlists")
+    results = spotify_session.user_playlists(config['spotify']['username'])
+    exclude_list = set([x.split(':')[-1] for x in config.get('excluded_playlists', [])])
+      
+    # get all the remaining playlists in parallel
+    if results['next']:
+        offsets = [ results['limit'] * n for n in range(1, math.ceil(results['total']/results['limit'])) ]
+        extra_results = await atqdm.gather( *[asyncio.to_thread(spotify_session.user_playlists, config['spotify']['username'], offset=offset) for offset in offsets ] )
+        for extra_result in extra_results:
+            playlists.extend([p for p in extra_result['items'] if p['owner']['id'] == config['spotify']['username'] and not p['id'] in exclude_list])
     return playlists
 
-def get_playlists_from_config(config):
+def get_playlists_from_config(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, config):
     # get the list of playlist sync mappings from the configuration file
-    return [(item['spotify_id'], item['tidal_id']) for item in config['sync_playlists']]
+    def get_playlist_ids(config):
+        return [(item['spotify_id'], item['tidal_id']) for item in config['sync_playlists']]
+    output = []
+    for spotify_id, tidal_id in get_playlist_ids(config):
+        try:
```

Comment:
Refactored this logic out of `sync_playlist()` so that we can pass the playlist objects directly rather than the playlist IDs